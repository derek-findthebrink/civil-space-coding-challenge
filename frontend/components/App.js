/* eslint-disable react/jsx-props-no-spreading */
import React from "react";
import { BrowserRouter, Route, Switch } from "react-router-dom";
import { ApolloProvider } from "@apollo/react-hooks";
import { Helmet } from "react-helmet";
import { client } from "../graphql/client";

// Components
import Home from "../features/home/components/Home";
import ArticlesContainer from "../features/articles/components/ArticlesContainer";
import ArticleContainer from "../features/articles/components/ArticleContainer";
import { Layout } from "./Layout";

// Routes
const ROOT = "/";
const ARTICLES = "/articles";
const ARTICLE = "/article/:id";

export const Routes = () => (
  <ApolloProvider client={client}>
    <Switch>
      <Route
        strict
        path={ARTICLE}
        render={(props) => <ArticleContainer {...props} />}
      />
      <Route
        strict
        path={ARTICLES}
        render={(props) => <ArticlesContainer {...props} />}
      />
      <Route strict path={ROOT} render={(props) => <Home {...props} />} />
    </Switch>
  </ApolloProvider>
);

export const App = () => {
  return (
    <>
      <Helmet>
        <link rel="preconnect" href="https://fonts.gstatic.com" />
        <link
          href="https://fonts.googleapis.com/css2?family=Source+Serif+Pro&family=Open+Sans:wght@400;700&display=swap"
          rel="stylesheet"
        />
      </Helmet>
      <Layout>
        <BrowserRouter>
          <Routes />
        </BrowserRouter>
      </Layout>
    </>
  );
};

export default App;
