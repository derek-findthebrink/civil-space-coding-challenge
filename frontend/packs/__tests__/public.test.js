import * as React from "react";
import { render, cleanup } from "@testing-library/react";
import App from "../public";

afterEach(cleanup);

it("matches snapshot", () => {
  const { asFragment } = render(
    <App />
  );
  expect(asFragment()).toMatchSnapshot();
});
