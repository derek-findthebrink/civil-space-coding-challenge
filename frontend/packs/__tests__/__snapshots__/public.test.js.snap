// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <main>
    <div
      class="layout__accent-bar"
    />
    <h1>
      Welcome!
    </h1>
    <div>
      <a
        href="/articles"
      >
        View Articles
      </a>
    </div>
  </main>
</DocumentFragment>
`;
